test/unit/gateways/checkout_v2_test.rb

```diff
@@ -0,0 +1,330 @@
+    assert response.test?
+  end
+
+  def test_failed_purchase
+    response = stub_comms do
+      @gateway.purchase(@amount, @credit_card)
+    end.respond_with(failed_purchase_response)
+    assert_failure response
+    assert_equal Gateway::STANDARD_ERROR_CODE[:card_declined], response.error_code
+  end
+
+  def test_successful_authorize_and_capture
+    response = stub_comms do
+      @gateway.authorize(@amount, @credit_card)
+    end.respond_with(successful_authorize_response)
+
+    assert_success response
+    assert_equal "charge_test_941CA9CE174U76BD29C8", response.authorization
+
+    capture = stub_comms do
+      @gateway.capture(@amount, response.authorization)
+    end.respond_with(successful_capture_response)
+
+    assert_success capture
+  end
+
+  def test_failed_authorize
+    response = stub_comms do
+      @gateway.authorize(@amount, @credit_card)
+    end.respond_with(failed_authorize_response)
+
+    assert_failure response
+    assert_equal "Invalid Card Number", response.message
+    assert response.test?
+  end
+
+  def test_failed_capture
+    response = stub_comms do
+      @gateway.capture(100, "")
+    end.respond_with(failed_capture_response)
+
+    assert_failure response
+  end
+
+  def test_successful_void
+    response = stub_comms do
+      @gateway.authorize(@amount, @credit_card)
+    end.respond_with(successful_authorize_response)
+
+    assert_success response
+    assert_equal "charge_test_941CA9CE174U76BD29C8", response.authorization
+
+    void = stub_comms do
+      @gateway.void(response.authorization)
+    end.respond_with(successful_void_response)
+
+    assert_success void
+  end
+
+  def test_failed_void
+    response = stub_comms do
+      @gateway.void("5d53a33d960c46d00f5dc061947d998c")
+    end.respond_with(failed_void_response)
+
+    assert_failure response
+  end
+
+  def test_successful_refund
+    response = stub_comms do
+      @gateway.purchase(@amount, @credit_card)
+    end.respond_with(successful_purchase_response)
+
+    assert_success response
+    assert_equal "charge_test_941CA9CE174U76BD29C8", response.authorization
+
+    refund = stub_comms do
+      @gateway.refund(@amount, response.authorization)
+    end.respond_with(successful_refund_response)
+
+    assert_success refund
+  end
+
+  def test_failed_refund
+    response = stub_comms do
+      @gateway.refund(nil, "")
+    end.respond_with(failed_refund_response)
+
+    assert_failure response
+  end
+
+  def test_successful_verify
+    response = stub_comms do
+      @gateway.verify(@credit_card)
+    end.respond_with(successful_authorize_response, failed_void_response)
+    assert_success response
+    assert_equal "Succeeded", response.message
+  end
+
+  def test_failed_verify
+    response = stub_comms do
+      @gateway.verify(@credit_card)
+    end.respond_with(failed_authorize_response, successful_void_response)
+    assert_failure response
+    assert_equal "Invalid Card Number", response.message
+  end
+
```

Comment:
It would cool to have a unit test checking the behavior of us getting back invalid json from the gateway.